M2/Macaulay2/m2/texhtml.m2

```diff
@@ -20,9 +20,9 @@ html TEX := str -> (
      	  f(///(^|[^\$])\$\$([^$]*[^\$])?\$\$([^$]|$)///,///\1</p><div style="text-align:center"><i>\2</i></div><p>\3///);
      	  f(///(^|[^\$])\$([^$]*[^\$])\$([^$]|$)///,///\1<i>\2</i>\3///);
 	  );
-     if match(///(^|[^\])\$///,str) then error("unmatched dollar signs in TeX string ",abbrev());
-     f(///\\\{///,///\lbrace ///);
-     f(///\\\}///,///\rbrace ///);
+     if match(///(^|[^\\])\$///,str) then error("unmatched dollar signs in TeX string ",abbrev());
+     f(///\\\{///,///\\lbrace ///);
+     f(///\\\}///,///\\rbrace ///);
```

Comment:
Why must you now put two backslashes here?   That's confusing.

Reply:
I think "\r" and "\l" are meaningful to Boost, similar to "\n". GNU regex didn't do any escaped character processing.

Reply:
Ah, no, sorry, it's because `\` itself is special, and if the next character isn't a known special character, the `\` is ignored. Here is what goes wrong:
```m2
replace(///\\l///, ">>\\l<<", "\\l") -- >><<
replace(///\l///, ">>\\l<<", "\\l")  -- \>><<
replace(///l///, ">>\\l<<", "\\l")   -- \>><<
```

Reply:
Ah, yes:
```m2
old way:

    i21 : replace("a","\\b","a")

    o21 = \b

new way:

    i13 : replace("a","\\b","a")

    o13 = b

    i14 : replace("a","\\\\b","a")

    o14 = \b
```

Perhaps this is worth warning the user about.

Reply:
This is mentioned both in changes.m2 and documentation for regular expressions. Do you have something else in mind?

Reply:
okay